mm2src/coins/eth/web3_transport/websocket_transport.rs

```diff
@@ -46,13 +46,13 @@ pub struct WebsocketTransport {
     node: WebsocketTransportNode,
     event_handlers: Vec<RpcTransportEventHandlerShared>,
     pub(crate) proxy_sign_keypair: Option<Keypair>,
-    controller_channel: Arc<ControllerChannel>,
+    controller_channel: ControllerChannel,
     connection_guard: Arc<AsyncMutex<()>>,
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 struct ControllerChannel {
-    tx: Arc<AsyncMutex<UnboundedSender<ControllerMessage>>>,
+    tx: UnboundedSender<ControllerMessage>,
     rx: Arc<AsyncMutex<UnboundedReceiver<ControllerMessage>>>,
 }
```

Comment:
we should wrap `ControllerChannel` in `Arc` as I assume it's better than implementing `Clone` because it ensures shared access to the same data through `Arc` rather than creating independent copies of each field when `Cloned` even tho the fields might be optimized for `Clone` we'll still end up with copies for each field.

Reply:
[8ae149b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2291/commits/8ae149bc194c4bab48a146a8b373e9a74e52306e)